### PR TITLE
LOVE-1342 add docs for volumes

### PR DESCRIPTION
**Pull request by deepu105**
## Definition of Done

**General**
 - [x] Branch is up-to-date with master
 - [x] Code is reviewed and tested by someone in the Kube Love team

### TODO:
see how to handle `GENERATE_XL_CONFIG` variable in `generate_node_conf` method of run script

### CODE REVIEW

File: docs/volumes.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+     - ~/XebiaLabs/xl-deploy-server/conf:/opt/xebialabs/xl-deploy-server/conf
+     - ~/XebiaLabs/xl-deploy-server/hotfix/lib:/opt/xebialabs/xl-deploy-server/hotfix/lib
+     - ~/XebiaLabs/xl-deploy-server/hotfix/plugins:/opt/xebialabs/xl-deploy-server/hotfix/plugins
+     - ~/XebiaLabs/xl-deploy-server/ext:/opt/xebialabs/xl-deploy-server/ext
+     - ~/XebiaLabs/xl-deploy-server/plugins:/opt/xebialabs/xl-deploy-server/plugins
+     - ~/XebiaLabs/xl-deploy-server/repository:/opt/xebialabs/xl-deploy-server/repository
+     - ~/XebiaLabs/xl-deploy-server/repository:/opt/xebialabs/xl-deploy-server/export
+     - ~/XebiaLabs/xl-deploy-server/repository:/opt/xebialabs/xl-deploy-server/work
+    environment:
+     - ADMIN_PASSWORD=admin
```
Comment (eltonkevani) [inline docs/volumes.md]: are you sure ADMIN_PASSWORD will work if GENERATE_XL_CONFIG is  false?

File: docs/volumes.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+**${APP_HOME}/hotfix/plugins** - This directory contains the hotfixes for the plugins used by XL Deploy. When instructed by XebiaLabs support personnel, you should drop any delivered plugin hotfixes in here.
+
+**${APP_HOME}/ext** - This directory contains the developed (exploded plugins) for XL Deploy. You can customize an existing plugin here, for example you can modify the `synthetic.xml` for a specific plugin here.
+
+**${APP_HOME}/plugins** - This directory contains the plugins that are running in XL Deploy. When mounted only plugins present in this directory will be loaded and you need to ensure to provide all required base plugins here. If you mount an empty directory the default bundled plugins will be loaded.
+
+**${APP_HOME}/repository** - This directory contains the embedded repository database for XL Deploy. Using the configuration injected through the ${APP_HOME}/conf volume, it is possible to configure the used database to a remotely running database engine, instead of using the embedded memory based one.
```
Comment (eltonkevani) [inline docs/volumes.md]: we should discourage users using local db and repository folder. this is only for testing only

File: docs/volumes.md (hunk 1 above)
Comment (deepu105) [reply docs/volumes.md]: ADMIN_PASSWORD is not set in the place where we check for `GENERATE_XL_CONFIG`, anyway let me try it once again with a different password

File: docs/volumes.md (hunk 2 above)
Comment (deepu105) [reply docs/volumes.md]: Ok i'll add note

**Pull request merged**